lighthouse-core/gather/driver.js

```diff
@@ -277,9 +277,17 @@ class Driver {
     return new Promise((resolve, reject) => {
       this.sendCommand('Page.getAppManifest')
         .then(response => {
-          // We're not reading `response.errors` however it may contain critical and noncritical
-          // errors from Blink's manifest parser:
-          //   https://chromedevtools.github.io/debugger-protocol-viewer/tot/Page/#type-AppManifestError
+          // handle errors
+          if (response.errors) {
+            response.errors.forEach(error => {
+              log.warn('driver', 'Manifest error', error.message);
+            });
+            const criticalError = response.errors.find(error => error.critical);
+            if (criticalError) {
+              return reject(new Error(`Manifest error: ${criticalError.message}`));
```

Comment:
while you're in here, do you want to flatten this promise? Rather than wrapping, can just `return this.sendCommand()...` and these `return reject(new Error())`s can just be `throw new Error()`